### DISCUSSION TITLE
dynamic library or module

### DISCUSSION THREAD

**Discussion by yanv1991** [vercel/next.js#12996] [Ideas]
Is there a way we can have dynamic modules, like https://loadable-components.com/docs/library-splitting/

I'm not sure if the actual implementation of dynamic is limited just for components

Thank you

**Comment by jamesmosier** [comment]
Next [has dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import)

```jsx
import dynamic from 'next/dynamic'

const DynamicComponent = dynamic(() => import('../components/hello'))

function Home() {
  return (
    <div>
      <Header />
      <DynamicComponent />
      <p>HOME PAGE is here!</p>
    </div>
  )
}

export default Home```

**Reply by yanv1991** [reply]
I know, but Does dynamic work for modules? For instance if you want to import moment.js dynamically

**Reply by jamesmosier** [reply]
Oh sorry, I misunderstood. It can work in a way:

```jsx
const DynamicComponent = dynamic({
  loader: () => import('luxon'),
  render(props, luxon) {
    console.log(luxon.DateTime);
    return <div>hello</div>;
  },
});

function Home() {
  return <DynamicComponent />;
}
```

To my knowledge, `next/dynamic` must return a component:
